src/x/server/server.go

```diff
@@ -140,11 +156,78 @@ func (s *server) Serve(l net.Listener) error {
 	return nil
 }
 
+func (s *server) isTLSConfigCacheValid() bool {
+	if s.tlsConfigCache == nil || s.certsTTLTicker == nil {
+		return false
+	}
+	select {
+	case <-s.certsTTLTicker.C:
+		return false
+	default:
+		return true
+	}
+}
+
+func (s *server) getTLSConfig() (*tls.Config, error) {
+	s.Lock()
+	defer s.Unlock()
+	if s.isTLSConfigCacheValid() {
+		return s.tlsConfigCache, nil
+	}
+	if s.tlsOpts.ClientCAFile() != "" {
```

Comment:
Instead of doing this flow per request, what do you think about having a background goroutine do the updates instead?

That is:

```
background goroutine: every N seconds, refresh the TLS config

request goroutines: read the current TLS config
```

Reply:
Fixed